src/validation.cpp

```diff
@@ -3097,6 +3097,13 @@ CBlockIndex* BlockManager::AddToBlockIndex(const CBlockHeader& block)
         pindexNew->nHeight = pindexNew->pprev->nHeight + 1;
         pindexNew->BuildSkip();
     }
+
+    if (pindexNew->pprev) {
+        for (unsigned i = 0; i < NUM_ALGOS_IMPL; i++)
+            pindexNew->lastAlgoBlocks[i] = pindexNew->pprev->lastAlgoBlocks[i];
```

Comment:
this loop is useless - if "i" is not the GetAlgo() there are only null ptr assignments

Reply:
and by the way - its not best practice to populate a array field of another instance on a foreign method